auth/presentation/src/main/java/com/nibalk/tasky/auth/presentation/register/RegisterScreen.kt

```diff
@@ -0,0 +1,171 @@
+        Spacer(modifier = Modifier.height(MaterialTheme.spacing.spaceMedium))
+        TaskyPasswordTextField(
+            state = state.password,
+            isPasswordVisible = state.isPasswordVisible,
+            onTogglePasswordVisibility = {
+                onAction(RegisterAction.OnTogglePasswordVisibilityClick)
+            },
+            hint = stringResource(id = R.string.auth_password),
+            modifier = Modifier.fillMaxWidth()
+        )
+        // Password Requirements
+        Spacer(modifier = Modifier.height(MaterialTheme.spacing.spaceMedium))
+        PasswordRequirements(
+            passwordValidationState = state.passwordValidationState
+        )
+        // Action Button
+        Spacer(modifier = Modifier.height(MaterialTheme.spacing.spaceLarge))
+        TaskyActionButton(
+            text = stringResource(id = R.string.auth_get_started).uppercase(),
+            isLoading = state.isRegistering,
+            enabled = state.canRegister,
+            modifier = Modifier.fillMaxWidth(),
+            onClick = {
+                onAction(RegisterAction.OnRegisterClick)
+            }
+        )
+        // Back Button
+        Spacer(modifier = Modifier.height(MaterialTheme.spacing.spaceExtraLarge))
+        Column(
+            modifier = Modifier.fillMaxSize(),
+            verticalArrangement = Arrangement.Bottom
+        ) {
+            TaskyBackButton(
+                isVisible = WindowInsets.ime.getBottom(LocalDensity.current) <= 0,
+                modifier = Modifier
+                    .padding(
+                        bottom = with(LocalDensity.current) {
+                            WindowInsets.navigationBars.getBottom(this).toDp()
+                        },
+                    )
+            )
+        }
+    }
+}
+
+@Composable
+private fun PasswordRequirements(
+    passwordValidationState: PasswordValidationState
+) {
+    TaskyPasswordRequirement(
+        text = stringResource(
+            id = R.string.auth_at_least_x_characters,
+            UserDataValidator.MIN_PASSWORD_LENGTH
+        ),
+        isValid = passwordValidationState.hasMinLength
+    )
+    Spacer(modifier = Modifier.height(MaterialTheme.spacing.spaceExtraSmall))
+    TaskyPasswordRequirement(
+        text = stringResource(
+            id = R.string.auth_at_least_one_number,
+        ),
+        isValid = passwordValidationState.hasNumericCharacter
+    )
+    Spacer(modifier = Modifier.height(MaterialTheme.spacing.spaceExtraSmall))
+    TaskyPasswordRequirement(
+        text = stringResource(
+            id = R.string.auth_contains_lowercase_char,
+        ),
+        isValid = passwordValidationState.hasLowerCaseCharacter
+    )
+    Spacer(modifier = Modifier.height(MaterialTheme.spacing.spaceExtraSmall))
+    TaskyPasswordRequirement(
+        text = stringResource(
+            id = R.string.auth_contains_uppercase_char,
+        ),
+        isValid = passwordValidationState.hasUpperCaseCharacter
+    )
+}
```

Comment:
Someone apparently adjusted the requirements a bit to match the ones from Runique? 👀

Reply:
😆  yeah, since I couldn't think of a better way to show the password requirements, and to validate the password while user is till typing it, I simply followed the Runique :)

Btw, did I miss any register screen requirements ? 🤔 💭